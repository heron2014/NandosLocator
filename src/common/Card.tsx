import React, { FC } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Pressable,
  GestureResponderEvent,
} from 'react-native';
import { RestaurantItem } from '../api';
import { colors, typography } from '../styles';

export interface ListItem {
  item: RestaurantItem;
  onPress: (event: GestureResponderEvent) => void;
}

const Card: FC<ListItem> = ({ item, onPress }) => {
  const handleStyleOnPress = ({ pressed }: { pressed: boolean }) => ([
    { backgroundColor: pressed ? colors.PRIMARY : colors.SECONDARY},
    styles.button,
  ]);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{item.name}</Text>
      <Text style={styles.text}>{item.geo.address.addressLocality}</Text>
      <Text style={styles.text}>{item.geo.address.streetAddress}</Text>
      <Text style={styles.text}>{item.geo.address.postalCode}</Text>
      <Pressable onPress={onPress} style={handleStyleOnPress}>
        <Text style={styles.text}>Check it out</Text>
      </Pressable>
    </View>
  );
};

Card.displayName = 'Card';

const styles = StyleSheet.create({
  container: {
    padding: 10,
    borderRadius: 4,
    marginBottom: 20,
    backgroundColor: colors.WHITE,
    alignItems: 'center',
  },
  title: {
    fontFamily: typography.FONT_FAMILY_BOLD,
    color: colors.BLACK,
    paddingVertical: 5,
    fontWeight: 'bold',
  },
  text: {
    fontFamily: typography.FONT_FAMILY_REGULAR,
    textAlign: 'center',
  },
  button: {
    marginVertical: 5,
    borderRadius: 4,
    height: 40,
    width: '70%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
export default Card;
