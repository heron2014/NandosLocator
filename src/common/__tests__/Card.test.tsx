import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';

test('renders correctly', () => {
  const item = {
    name: 'Nandos name',
    url: 'https://someLink',
    geo: {
      address: {
        streetAddress: 'High Street',
        addressLocality: 'Address locality',
        postalCode: 'N1 H9',
      },
    },
  };

  const tree = renderer
    .create(<Card item={item} onPress={() => null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
