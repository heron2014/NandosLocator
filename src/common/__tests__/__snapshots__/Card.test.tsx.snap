// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "marginBottom": 20,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "AvenirNext-Bold",
        "paddingVertical": 5,
      }
    }
  >
    Nandos name
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-Medium",
        "textAlign": "center",
      }
    }
  >
    Address locality
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-Medium",
        "textAlign": "center",
      }
    }
  >
    High Street
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-Medium",
        "textAlign": "center",
      }
    }
  >
    N1 H9
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#ffff00",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "height": 40,
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "70%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Avenir-Medium",
          "textAlign": "center",
        }
      }
    >
      Check it out
    </Text>
  </View>
</View>
`;
